public/app/core/components/PasswordField/PasswordField.tsx

```diff
@@ -0,0 +1,37 @@
+import React, { FC, useState } from 'react';
+import { selectors } from '@grafana/e2e-selectors';
+
+import { Input, IconButton } from '@grafana/ui';
+
+export interface Props {
+  autoFocus?: boolean;
+  autoComplete?: string;
+  id?: string;
+  passwordHint?: string;
+}
+
+export const PasswordField: FC<Props> = ({ autoComplete, autoFocus, id, passwordHint, ...props }) => {
+  const [showPassword, setShowPassword] = useState(false);
+
+  return (
+    <Input
+      id={id}
+      autoFocus={autoFocus}
+      autoComplete={autoComplete}
+      {...props}
+      type={showPassword ? 'text' : 'password'}
+      placeholder={passwordHint}
+      aria-label={selectors.pages.Login.password}
+      suffix={
+        <IconButton
+          name={showPassword ? 'eye-slash' : 'eye'}
+          surface="header"
```

Comment:
```suggestion
          surface="header"
          aria-controls={id}
          aria-expanded={showPassword}
```
I think it'd be a good idea to add the aria attributes to the toggle button for improving its accessibility.

Reply:
addressed